frontend/src/lib/inferArgSig.ts

```diff
@@ -102,12 +102,24 @@ export function argSigToJsonSchemaType(
 		} else if (t.list && typeof t.list == 'object' && 'str' in t.list) {
 			newS.items = { type: 'string', enum: t.list.str }
 		} else {
-			newS.items = { type: 'object' }
+			let resource: string | undefined = undefined
+
+			if (t.list && typeof t.list == 'object' && 'resource' in t.list && t.list.resource) {
+				resource = t.list.resource as string
+			}
+
+			if (resource) {
+				newS.items = { type: 'resource', resource: resource }
+			} else {
+				newS.items = { type: 'object' }
+			}
```

Comment:
seems convoluted, why not just add more cases to the top elseif

Reply:
Done